Feature electric weights

mclarke2 opened the pull request.

MGuidotti committed:
Update aircraft_aerodynamic_analysis.py

ssankalp26 committed:
increase transition test tolerence for windows machines

Co-Authored-By: Dr. Matthew Clarke <25692010+mclarke2@users.noreply.github.com>

ssankalp26 committed:
Removed python 3.8 compatibility since vtk doesnt exist for mac os with python 3.8 and also it is old

ARmolloy committed:
Electric Transport Weight Buildup

Combines elements from electric GA, conventional transport, and EVTOL

ARmolloy committed:
Regression Electric Transport

ARmolloy committed:
Piston Engine Fixes

These changes were previously on the bug-piston_engine branch. That branch will be deleted

ssankalp26 committed:
weights update

ssankalp26 committed:
Merge branch 'develop' into feature-Electric_Weights

ssankalp26 committed:
Merge branch 'develop' into feature-Electric_Weights

ssankalp26 committed:
Fix cargo and systems weight calculations

Cast cargo mass to int for container weight calculation in compute_operating_items_weight. Remove unnecessary conversion of wing sweep and simplify furniture weight formula in compute_systems_weight.

ssankalp26 committed:
Refactor fuel tank and mass property handling

Assigns Liquid Hydrogen propellant to the tank class, updates fuel mass assignment in compute_fuel_volume, improves structural optimization bounds, increases interpolation resolution for tank volume, and refines logic for max fuel and zero fuel mass estimation in mass properties preprocessing.

ARmolloy committed:
Merge remote-tracking branch 'origin/bug_fix-aft_tank_volume_computation' into feature-Electric_Weights

ssankalp26 committed:
Merge branch 'bug_fix-aft_tank_volume_computation' into feature-Electric_Weights

ssankalp26 committed:
plot fixes for fuselage tanks

ssankalp26 committed:
Refactor fuel volume computation and update Hydrogen BWB structure

Replaces 'update_fuel_volume' with 'compute_fuel_volume' and adds 'update_max_fuel' in geometry and related methods for clarity and expanded control. Refactors function signatures and usage throughout the codebase. Moves Hydrogen BWB weight buildup methods to a new Semi_Empirical subpackage and updates imports accordingly. Adds new __init__.py files for package structure and improves fuel system weight calculation logic.

ssankalp26 committed:
Fix distributor and bus handling in coolant line setup

Updated Coolant_Line to correctly iterate over distributor objects and handle battery modules. Adjusted wavy_channel_geometry_setup to pass a list of buses to Coolant_Line, ensuring compatibility with the revised constructor logic. Also fixed indentation in Network power computation for clarity.

ssankalp26 committed:
Update append_battery_conditions.py

ssankalp26 committed:
Update append_battery_conditions.py

ssankalp26 committed:
Update append_battery_conditions.py

ssankalp26 committed:
Update append_battery_conditions.py

ssankalp26 committed:
Rename 'aile_width' to 'aisle_width' in cabin classes

Corrects the spelling of 'aile_width' to 'aisle_width' across Business, Economy, and First cabin classes, related geometry functions, and vehicle setup scripts for consistency and clarity.

ssankalp26 committed:
Update air cooling and coolant line handling

Adjusted default convective heat transfer coefficient and removed flowspeed from Air_Cooled battery component. Modified air_cooled_performance to compute coolant velocity from freestream conditions. Updated Electric_Twin_Otter vehicle setup to pass bus as a list to Coolant_Line. Commented out fuel assertion in mass_properties preprocessing.

ARmolloy committed:
Raymer electric semi-empirical transport

ARmolloy committed:
Update mass_properties.py

Including ability to use a user-defined CG for moment of inertia calculations

ssankalp26 committed:
Update compute_segment_volume.py

ARmolloy committed:
Update generate_3d_fuel_tank_points.py

ARmolloy committed:
Inertia Tensor Bug Fix

Fixed issue with the parallel axis theorem and incorrect numpy usage

ARmolloy committed:
Setting for Compute Neutral Point

ARmolloy committed:
BWB CG updates

ARmolloy committed:
Update wing_planform.py

ARmolloy committed:
BWB CG Updates

ARmolloy committed:
Merge remote-tracking branch 'origin/feature-CG_Temp' into feature-Electric_Weights

ARmolloy committed:
Update compute_vehicle_center_of_gravity.py

ssankalp26 committed:
Update compute_vehicle_center_of_gravity.py

ssankalp26 committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

ssankalp26 committed:
Refine mass property calculations and clean up debug code

Removed addition of hydraulics and electrical systems from center body mass calculation in compute_vehicle_center_of_gravity.py. Also removed unused debug code and commented out print statements in mass_and_intertia_functions.py to clean up output.

ARmolloy committed:
Weights

ARmolloy committed:
Merge remote-tracking branch 'origin/develop' into feature-Electric_Weights

ARmolloy committed:
Update compute_segment_volume.py

ARmolloy committed:
Regression Updates

ARmolloy committed:
Update compute_segment_volume.py

ARmolloy committed:
Update VERSION

ARmolloy committed:
Update VERSION

New new version. Incremented to .21 from .19 to account for incoming .20 version

ARmolloy committed:
Update operating_empty_weight_test.py

ssankalp26 committed:
Merge branch 'develop' into feature-Electric_Weights

ARmolloy committed:
Update operating_empty_weight_test.py

Adding Hydrogen BWB

ssankalp26 committed:
Update generate_3d_fuel_tank_points.py

ssankalp26 committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

ARmolloy committed:
Regression Updates and NP bug Fix

Also changed the BWB payload weight to be less than max payload weight.

ssankalp26 committed:
Add hydrogen BWB weight results and update test

Added four new .res files for hydrogen BWB weight analysis (PERSUS/Non-PERSUS, Simple/Complex). Updated operating_empty_weight_test.py to use compute_fuel_volume and revised fuel tank setup for hydrogen BWB tests. Set update_regression_values to False for regression stability.

ssankalp26 committed:
Update compute_liquid_hydrogen_tank_volume.py

ssankalp26 committed:
it is cargo mass and not cargo bay mass, dont change it back

ssankalp26 committed:
removed redundant weight reduction factors

ssankalp26 committed:
it is thrust reverser

ssankalp26 committed:
empenage differentiation

ARmolloy committed:
Component weight updates

Adds and modifies the component mass_properties.mass variables

ssankalp26 committed:
lopa revert

it works for now, will correct stuff later after scitech

ssankalp26 committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

ssankalp26 committed:
Fix insulation volume calculation formula

Corrected the formula for insulation volume by removing redundant parentheses and ensuring proper multiplication by pi in the spherical term. This improves the accuracy of the tank insulation volume computation.

MGuidotti committed:
Refactor cabin length calculation in LOPA functions

Updated functions to pass and update cabin_length explicitly during seat map layout computation. This improves clarity and ensures accurate cabin length assignment for each cabin in the layout process.

mclarke2 committed:
Merge branch 'develop' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

mclarke2 committed:
Fix variable assignments and improve tank volume calculation

Corrected a variable name typo in LOPA layout assignment and fixed attribute access for electrical systems in operating empty weight calculation. Updated the integral tank volume computation to use more precise percent chord start/end attributes, improving accuracy in wing tank geometry calculations.

ARmolloy committed:
Update aircraft_aerodynamic_analysis.py

ARmolloy committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

mclarke2 committed:
Refactor preprocess routines to use analyses object

Updated geometry and mass properties preprocess routines to accept the full analyses object instead of separate settings and vehicle arguments. Adjusted function calls and signatures throughout related modules for consistency and improved maintainability.

mclarke2 committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

mclarke2 committed:
Refactor cabin passenger logic and update tests

Updated cabin passenger assignment to use number_of_seats instead of number_of_passengers in load and trim diagram calculations. Improved handling of blended wing body cabins. Refactored geometry preprocessing to remove redundant passenger summation. Updated aircraft aerodynamics test to use new vehicle and config setup functions, and added base analysis setup routines for test consistency.

MGuidotti committed:
Update geometry.py

mclarke2 committed:
Refactor cargo bay container handling and passenger count

Added a container attribute to Cargo_Bay and updated operating items weight calculations to handle container mass per cargo bay. Improved passenger count logic in load and trim diagram computation to use number_of_seats instead of number_of_passengers for more accurate simulation. Minor cleanups in payload weight distribution and cabin offset handling.

mclarke2 committed:
Update compute_load_and_trim_diagram.py

mclarke2 committed:
Refactor fuel volume and neutral point settings usage

Replaces 'overwrite_fuel_volume' with 'compute_fuel_volume' in geometry settings and ensures 'compute_neutral_point' is set appropriately in stability settings. Updates related analysis and test files to use the new settings, improving clarity and consistency in configuration management.

mclarke2 committed:
Merge branch 'feature-Electric_Weights' of https://github.com/leadsgroup/RCAIDE_LEADS into feature-Electric_Weights

mclarke2 committed:
Refactor passenger layout and fuselage geometry routines

Improved handling of passenger accommodation layout by refactoring logic in compute_layout_of_passenger_accommodations and extracting property computation to a new function. Removed unused cabin_offset attributes and streamlined geometry_preprocess_routine to always compute layouts and planforms. Minor formatting and import corrections applied for consistency.

mclarke2 committed:
Fix typo: correct 'intertia' to 'inertia' in comments and docs

Replaces all occurrences of the misspelled word 'intertia' with the correct spelling 'inertia' in comments, docstrings, and print statements across multiple mass properties and aerodynamics modules. This improves code readability and documentation accuracy.

mclarke2 committed:
Add 3D cargo bay plotting and improve payload logic

Introduces generate_3d_cargo_bay_points for visualizing cargo bays in 3D plots and integrates cargo bay rendering into plot_3d_vehicle. Refines payload and cargo mass assignment logic in compute_payload_weight for both common and Raymer methods, ensuring correct mass distribution and initialization. Sets default vehicle cargo mass to 0 in Vehicle_Mass_Properties.

mclarke2 committed:
Enhance load diagram analysis and plotting

Improved the computation of payload weights and mass properties for both general and blended wing body configurations. Refactored the load and trim diagram logic to better handle segment tags, discretization, and bounds, and added more flexible plotting options for axis and static margin limits. Fixed indentation and logic errors in mass properties preprocessing and updated plotting to allow custom axis and margin limits.

mclarke2 committed:
Update plot_3d_vehicle.py

mclarke2 committed:
Refactor nacelle handling and weight assignments

Standardizes nacelle attribute access across the codebase, removing checks for 'nacelle' in propulsor and using direct attribute access instead. Refactors weight assignment logic for propulsors and nacelles, and updates related drag and geometry calculations. Also introduces y_offset_distance for cabin classes and cleans up related code.

mclarke2 committed:
Update VERSION

mclarke2 committed:
Refactor propulsion system weight calculations

Improved handling of nacelle, thrust reverser, and miscellaneous weights in propulsion system calculations for FLOPS and Hydrogen BWB methods. Removed direct assignment of mass properties in Raymer operating empty weight calculation. Updated regression test flag and adjusted .res files to reflect new structural and propulsion breakdowns, including empennage and revised fuel system weights.

mclarke2 commented on RCAIDE/Framework/Networks/Network.py:
```diff
@@ -166,9 +166,9 @@ def evaluate(network,state,center_of_gravity):
-                # compute power from each componemnt 
-                conditions.energy.busses[bus.tag].power_draw        += (total_elec_power- state.conditions.energy.busses[bus.tag].regenerative_power*bus_voltage ) * bus.power_split_ratio  /bus.efficiency   
-                conditions.energy.busses[bus.tag].current_draw       = conditions.energy.busses[bus.tag].power_draw/bus_voltage  
+                # compute power from each componemnt
```
Typo fixed

mclarke2 commented on RCAIDE/Library/Methods/Mass_Properties/Center_of_Gravity/compute_vehicle_center_of_gravity.py:
```diff
@@ -108,6 +119,8 @@ def compute_vehicle_center_of_gravity(vehicle , update_center_of_gravity=True):
 
     for key in vehicle.keys():
         item = vehicle[key]
+        if key =='wings':
```
Removed

mclarke2 committed:
Refactor Boeing 787 payload range test and update configs

Refactored the Boeing 787 payload range validation test to remove the payload_range_test function and inline its logic in main, updating reference values for payload range results. Added analysis and mission setup utilities to the test file. Updated BWB and Boeing 787 vehicle definitions for payload and cargo bay mass. Fixed minor typos and improved error messages in import_vsp_vehicle.py. Corrected index selection in compute_airfoil_properties.py and removed unused code in compute_vehicle_center_of_gravity.py. Updated test values in blended wing body and aircraft aerodynamics tests for consistency.

mclarke2 committed:
Refactor array indexing and plotting routines

Updated multiple files to remove unnecessary float conversions and use direct array indexing for improved clarity and consistency. Replaced deprecated np.trapz with np.trapezoid in rotor noise calculations. Refactored rotor performance plotting to use matplotlib instead of plotly, and standardized legend and axis handling. Adjusted weight analysis result files to correct payload and operational item calculations.

mclarke2 committed:
Fix array indexing and shape handling in multiple modules

Corrects array indexing and shape handling in several files to ensure proper extraction of scalar values and broadcasting. Updates include returning scalar values from arrays, accessing elements with correct indices, and adjusting assignments to match expected array shapes. These changes improve compatibility and prevent potential runtime errors related to array dimensions.

mclarke2 committed:
Fix indexing for objective values and constraints in optimization modules

Updated several optimization and plotting modules to correctly index objective values and constraints, ensuring compatibility with returned data shapes. This addresses issues where returned values were nested arrays or lists, preventing assignment or comparison errors.

mclarke2 committed:
Fix indexing and return value issues in optimization and plotting

Corrected the use of the objective function's return value in Nexus.py to properly extract the first element. Updated the loop in plot_lift_distribution.py to use the correct index for the number of wings, preventing potential iteration errors.

ARmolloy committed:
Update Trust_Region_Optimization.py

Curiosity on if this fixes the Python indexing issue. if it does then I will revisit the rest of the script

ARmolloy committed:
Update Tutorials and NP default setting

NP does not need to be run for every single analysis so I've changed the default to False. This will reduce computational time

ARmolloy committed:
Update Vortex_Lattice_Method.py

Neutral point defaults to zero if it is not separately calculated

ARmolloy commented on RCAIDE/Library/Methods/Mass_Properties/Weight_Buildups/Conventional/BWB/FLOPS/compute_systems_weight.py:
```diff
@@ -182,7 +180,7 @@ def compute_systems_weight(vehicle):
     WAVONC  = 15.8 * DESRNG ** 0.1 * NFLCR ** 0.7 * FPAREA ** 0.43  # avionics weight
     
-    WFURN   = 127 * NFLCR + 112 *  NPF + 78 *  NPB + 44 * NPE  + 2.6 * (ACABIN*(WF+DF*NBAY)/WF + WF*DF*(1+1/np.cos(SWPLE)))
+    WFURN   = 127 * NFLCR + 112 *  NPF + 78 *  NPB + 44 * NPE
```
Can someone remind me what our thinking with deleting this section was?

ARmolloy replied:
To follow up on this, I don't remember the reasoning and I know this extra expression is in the actual FLOPS estimates.

ssankalp26 replied:
Not sure what happened, seems that the equation that existed for a special case equation meant for bwbs

ARmolloy replied:
I'd agree, taking a look at the FLOPS docuentation it does appear to be a special equation for HWB aircraft. In that case, I'd say we need to revert this change.

mclarke2 replied:
Reverted back to what is on the develop branch

ARmolloy commented on RCAIDE/Library/Methods/Mass_Properties/Weight_Buildups/Conventional/Transport/Raymer/compute_operating_empty_weight.py:
```diff
@@ -102,26 +79,7 @@ def compute_operating_empty_weight(vehicle, settings=None):
-        # for propulsor in network.propulsors:
-        #     if 'motor' in propulsor:                           
-        #         W_energy_network.W_motor +=  propulsor.motor.mass_properties.mass
-        #         W_energy_network_total  +=  propulsor.motor.mass_properties.mass
```
This should be removed

mclarke2 replied:
This has been removed. Not sure why it does not show

ssankalp26 commented on RCAIDE/Framework/Analyses/Stability/Vortex_Lattice_Method.py:
```diff
@@ -102,7 +102,10 @@ def __defaults__(self):
+        if self.settings.compute_neutral_point:
+            compute_neutral_point(self, vehicle)  
+        elif vehicle.neutral_point == None:
+            vehicle.neutral_point = 0.0          
```
Should we have an option of specifying a NP from the analyses, so that if we are doing multiple simulations with the same aircraft we can reuse the the NP

ARmolloy replied:
I believe this situation may already be covered, we'd just need to make sure that settings.compute_neutral_point = False, but then in that case it'd just use the vehicle's NP

mclarke2 replied:
I may actually make this more foolproof. and have a raise error instead of setting it to zero. As you guys pointed out, having this defaulted is a bad precedent

ssankalp26 commented on RCAIDE/Framework/Analyses/Weights/Hydrogen_BWB.py:
```diff
@@ -0,0 +1,52 @@
+        N/A
+        """ 
+        self.aircraft_type                                 = 'BWB' 
+        self.settings.PRSEUS                               = True
```
method default not specified

mclarke2 replied:
No need. it inherits from Hydrogen which as things specified

ssankalp26 commented on RCAIDE/Library/Components/Fuselages/Cabins/Classes/Business.py:
```diff
@@ -28,9 +28,11 @@ def __defaults__(self):
         self.seat_length                         = 25 *  Units.inches
         self.seat_pitch                          = 40 *  Units.inches
-        self.aisle_width                          = 15  *  Units.inches          
+        self.aisle_width                         = 15  *  Units.inches          
```
PLease plot a bwb, replicate the scitech bwb, with the new method for x offset and side cabin y offset to make sure it works properly, We could add some the lopa seat locations as a truth value in the regressions

mclarke2 replied:
Good idea, on using the LOPA coordinates and a verification.  Done in the BWB aerodynamics test

ssankalp26 commented on RCAIDE/Library/Methods/Geometry/LOPA/LOPA_functions.py:
```diff
@@ -184,84 +162,70 @@ def get_seat_x_coords(cabin,cabin_class,cabin_class_origin):
-    return s_x_coord , object_type, cabin_class_origin
- 
-
+    return s_x_coord , object_type, cabin_class_origin,cabin_length
```
these are the locations that can be used as truth values, if not atleast check if this works as it is expected to, the scitech BWB sould have 164 or something seats.

mclarke2 replied:
Resolved in previous comment

ssankalp26 commented on RCAIDE/Library/Methods/Geometry/LOPA/compute_layout_of_passenger_accommodations.py:
```diff
@@ -1,179 +1,152 @@
-        
+        seat_data[:, 3] += cabin.width / 2
+        seat_data  = update_seat_map_layout_using_cabin_taper(seat_data,cabin)
+        seat_data[:, 3] += side_cabin_offset  +  cabin_class.y_offset_distance
```
please verify

mclarke2 replied:
I checked this, basically it shifts the side cabin but a have a width since the center cabin is defined along the center axis.

ssankalp26 commented on RCAIDE/Library/Methods/Mass_Properties/Center_of_Gravity/compute_vehicle_center_of_gravity.py:
```diff
@@ -76,7 +76,18 @@ def compute_vehicle_center_of_gravity(vehicle , update_center_of_gravity=True):
+            wing.aft_center_body.tag = 'aft_center_body'
+            wing.center_body.origin = [[0.5*(wing.chords.root - wing.aft_center_body.length),0,0]]     
+            wing.center_body.tag = 'center_body'
+            wing.center_body.mass_properties.mass += vehicle.mass_properties.weight_breakdown.operational_items.total
```
Operational Items are being added twice

mclarke2 replied:
Corrected this, but we already plan to update this in the next PR

ssankalp26 commented on RCAIDE/Library/Methods/Mass_Properties/Weight_Buildups/Conventional/Common/compute_payload_weight.py:
```diff
@@ -91,13 +87,18 @@ def compute_payload_weight(vehicle, W_passenger=195 * Units.lbs, W_baggage=30 *
+        if cargo_bay.cargo.mass_properties.mass == 0: 
+            cargo_bay.cargo.mass_properties.mass   = vehicle.mass_properties.cargo * (cargo_bay_volume / total_volume)
+        else:
+            vehicle.mass_properties.cargo += cargo_bay.cargo.mass_properties.mass
```
need to verify vehicle.mass_properties.cargo is what is being used in weight buildup, Can be done by generating the weight build up report and comparing with the weight breakdown in the paper. Ensures consistency with what we have been reporting for AACES as well

mclarke2 replied:
@ARmolloy  I will leave you to do this next week, that said I think we should be good

ARmolloy replied:
In the current form, the else is causing issues becauses it's effectively adding on to the cargo weight in subsequent segments where the cargo mass has already been assigned to bays. i.e. for a 4 segment mission, the vehicle.mass_properties.cargo = 4000 where it started as 1000

ARmolloy replied:
I'll admit it's not the cleanest, but I think the edits I've added take care of the issue of the weight being added on in subsequent segments.

ARmolloy replied:
I checked results against the scitech bwb paper and most things agree except for furnishing weight. This is due to the added expression for BWBs (When I tried it without, it lined up). Previously, the operating items weight was off, BUT I fixed this by taking the cargo weight float to be an int() in the container weight section of computing operating items weight. Because we were having container weight added when the cargo mass was 2e-13 (this should be zero, it was a numerical carryover from elsewhere). Hope that makes sense

ssankalp26 commented on RCAIDE/Library/Methods/Mass_Properties/Weight_Buildups/Conventional/Common/compute_payload_weight.py:
```diff
@@ -91,13 +87,18 @@ def compute_payload_weight(vehicle, W_passenger=195 * Units.lbs, W_baggage=30 *
     for cargo_bay in vehicle.cargo_bays:
         total_volume += (cargo_bay.length * cargo_bay.width * cargo_bay.height)
-    
+     
```
What would happen if we defined a cargo bay, because that seems to be a requriement for load trim, would that mean that the weight breakdown for load trim and payload range would be different ?

mclarke2 replied:
You can do a load trim without a cargo bay.

mclarke2 replied:
I realized the issue with the logic. Thanks!

ssankalp26 commented on RCAIDE/Library/Methods/Mass_Properties/Weight_Buildups/Conventional/General_Aviation/Raymer/compute_payload_weight.py:
```diff
@@ -49,38 +49,39 @@ def compute_payload_weight(vehicle, W_passenger=195 * Units.lbs, W_baggage=30 *
         total_volume += (cargo_bay.length * cargo_bay.width * cargo_bay.height)
-    
+     
     for cargo_bay in vehicle.cargo_bays:
```
We should for sure not be including "cargo bays" in general aviation

mclarke2 replied:
agreed. Removed

ssankalp26 commented on RCAIDE/Library/Methods/Performance/compute_load_and_trim_diagram.py:
```diff
@@ -229,65 +237,96 @@ def compute_load_and_trim_diagram(mission = None, cruise_segment_tag = "cruise",
+                                        previous_cargo_bays = 0
+                                        for cargo_bay in vehicle.cargo_bays: 
+                                            for c_i in range(len(percent_cargo)): 
+                                                cargo_bay.cargo.mass_properties.mass     = percent_cargo[c_i] * vehicle_0.cargo_bays[cargo_bay.tag].cargo.mass_properties.mass 
```
Same questions for cargo bay, plese verify.

mclarke2 replied:
Good catch, resolved

ssankalp26 commented on RCAIDE/Library/Methods/Powertrain/Sources/Fuel_Tanks/Integral_Tank/compute_integral_tank_volume.py:
```diff
@@ -216,8 +216,39 @@ def compute_wing_integral_tank_volume(fuel_tank,wing):
+            (p4x + p1x)*(p4x*p1y - p1x*p4y)
+        )
+        
+        
```
I know i put this code in but please cleam this up, Thank you

mclarke2 replied:
Cleaned up

ssankalp26 reviewed (commented):
There were some changes made in the payload range script to enable the faster compute of the Hydrogen BWB, if those changes can be brough over and cleaned up it owuld be great

ssankalp26 commented on VnV/Validation/aircraft_performance/test_Boeing_787_payload_range.py:
```diff
@@ -29,11 +38,11 @@ def main():
-        "oew_plus_payload": np.array([160842.56285273, 160842.56285273, 126607.        , 116842.56285273]),
-        "fuel": np.array([     0.        ,  67087.43714727, 101323.        , 101323.        ]),
-        "takeoff_weight": np.array([     0.        , 227930.        , 227930.        , 218165.56285273]),
+        "range": np.array([       0.        , 10379861.95907129, 17639210.23636528, 18306191.89479472]),
+        "payload": np.array([44000.        , 44000.        ,  9526.30115302,     0.        ]),
+        "oew_plus_payload": np.array([161080.69884698, 161080.69884698, 126607.        , 117080.69884698]), 
+        "fuel": np.array([     0.        ,  66849.30115302, 101323.        , 101323.        ]),
+        "takeoff_weight": np.array([     0.        , 227930.        , 227930.        , 218403.69884698]),
     }
```
Why have these values changed, yes they are low but I suspect that weight build up values are different. PLease verify where these changes are coming from

ssankalp26 replied:
To addon to this, there are several correction factors for the weight build up as well, so it would be worth reviewing those as well

mclarke2 replied:
This will resolve itself once we address the other issue. Good catch

mclarke2 committed:
Update weight calculations and stability logic

Improves systems weight calculation for BWB vehicles, updates neutral point computation logic in stability analyses, refines center of gravity and payload computations, and adds more robust handling for cargo bays and load diagrams. Also updates verification tests and expected results to reflect these changes.

mclarke2 committed:
Update test_automatic_regression.py

ARmolloy committed:
Update compute_payload_weight.py

ARmolloy committed:
Convert cargo weight to int

In effect this is making sure that values that are pretty much 0 are 0

mclarke2 committed:
Updates to regression

ARmolloy reviewed (approved):
Comments have been addressed and everything seems good. Additional changes to CG and inertia tensor will be included in an upcoming PR.
